overlord/snapstate/snapstate.go

```diff
@@ -244,6 +244,12 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	addTask(startSnapServices)
 	prev = startSnapServices
 
+	if fromStore {
+		updateExtraInfo := st.NewTask("save-extra-info", fmt.Sprintf(i18n.G("Update cache of store-side info for %q"), snapsup.InstanceName()))
+		addTask(updateExtraInfo)
```

Comment:
Could you please apply the same trick Pawel did to a similar local `addTask` helper, namely that it sets the `prev` value internally. This is just so easy to miss with disastrous consequences.

Reply:
no (look at `prev` vs `chain`)